src/compiler/checker.ts

```diff
@@ -6699,7 +6699,13 @@ namespace ts {
                 let hasThisParameter: boolean;
                 const iife = getImmediatelyInvokedFunctionExpression(declaration);
                 const isJSConstructSignature = isJSDocConstructSignature(declaration);
-                const isUntypedSignatureInJSFile = !iife && !isJSConstructSignature && isInJavaScriptFile(declaration) && !hasJSDocParameterTags(declaration);
+                const isUntypedSignatureInJSFile = !iife &&
+                    isInJavaScriptFile(declaration) &&
+                    declaration.kind !== SyntaxKind.FunctionType &&

```

Comment:
Rather than skipping (some of the) the type-space signature declaration kinds, can we just check if it _is_ a concrete declaration kind? That is the point here, right; to exclude type-space declaration kinds?

```ts
export type ConcreteSignatureDeclaration =
        | FunctionDeclaration
        | MethodDeclaration
        | ConstructorDeclaration
        | AccessorDeclaration
        | FunctionExpression
        | ArrowFunction;

export function isConcreteSignatureDeclaration(declaration: Node): node is ConcreteSignatureDeclaration {
    return isFunctionExpression(declaration) || isArrowFunction(declaration) || isMethodOrAccessor(declaration) || isFunctionDeclaration(declaration) || isConstructorDeclaration(declaration);
}
```

Reply:
Yes, a type signature is by definition not untyped, even if it's in a JS file. I used your code in the new commit.